### ISSUE TITLE
bug: Shell integration is installed even with `programs.fzf.enableFishIntegration = false`

### ISSUE THREAD

**Issue by s3igo** [nix-community/home-manager#5904]
### Are you following the right branch?

- [X] My Nixpkgs and Home Manager versions are in sync

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue description

Even when setting `programs.fzf.enableFishIntegration = false`, the shell integration still gets installed in the fish environment. This seems to be caused by the [`pkgs.fzf`](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/by-name/fz/fzf/package.nix) installing the shell integration during the postInstall phase, regardless of the contents of programs.fish.interactiveShellInit.

I was able to fix this issue with the following configuration:

```nix
programs.fzf = {
  enable = true;
  enableFishIntegration = false;
  package = pkgs.fzf.overrideAttrs {
    # Prevent shell integrations from installing automatically
    postInstall = ''
      installManPage man/man1/fzf.1
    '';
  };
};
```

However, I am not sure if this is the appropriate way to fix the issue. Also, I have not yet tested whether the same problem occurs in bash or zsh.

### System information

```markdown
- system: `"aarch64-darwin"`
 - host os: `Darwin 23.6.0, macOS 14.6.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.7`
 - nixpkgs: `/nix/store/vyv637pn8fx0gs28pfqy5b3jkrmd7cvl-source`
```

**Comment by rycee** [comment]
Yeah, I think we cannot do much about this inside Home Manager. To make the shell integration optional it would require some support for that in the Nixpkgs package. E.g., an attribute such as `installShellIntegrations` or something. Then we could set that inside the module. On the other hand, that would force the user to recompile the package.

Unfortunately we cannot do the `overrideAttr` trick in Home Manager since we can't account for future changes in the package.
